src/asynciterable/operators/_flatten.ts

```diff
@@ -119,6 +121,8 @@ export class FlattenConcurrentAsyncIterable<TSource, TResult> extends AsyncItera
             }
             case Type.INNER: {
               --active;
+              // add the outer iterator to the race
+              results[0] = outer.next();
```

Comment:
I don't think this is safe, since here we are still potentially waiting on the latest promise from the outer iterator to resolve.

Reply:
I see what you mean. I'll see if I can't figure out a better way of doing it without potentially loosing values.

Reply:
Hey @trxcllnt thanks for pointing this out, https://github.com/ReactiveX/IxJS/pull/376/commits/8f90bc7d19e0ff53559ab6996619652f5017f292 resolves the tests locally for me, as you Identified some were failing because I was loosing items due to the above and others were failing because next was called on the completed outer above.